ai-data/generative-apis/how-to/query-text-models.mdx

```diff
@@ -0,0 +1,164 @@
+
+The [Chat API](/ai-data/generative-apis/api-cli/using-chat-api/) is an OpenAI-compatible REST API for generating and manipulating conversations.
+
+You can query the models programmatically using your favorite tools or languages.
+In the following example, we will use the OpenAI Python client.
+
+### Installing the OpenAI SDK
+
+You can install it using pip:
+
+```bash
+pip install openai
+```
+
+### Initializing the client
+
+Initialize the OpenAI client with your base URL and API key:
+
+```python
+from openai import OpenAI
+
+# Initialize the client with your base URL and API key
+client = OpenAI(
+    base_url="https://api.scaleway.ai/v1",  # Scaleway's Generative APIs service URL
+    api_key="<SCW_API_KEY>"  # Your unique API key from Scaleway
+)
+```
+
+### Generating a chat completion
+
+You can now create a chat completion, for example for example with the `llama-3.1-8b-instruct` model:
+
+```python
+# Create a chat completion using the 'llama-3.1-8b-instruct' model
+response = client.chat.completions.create(
+    model="llama-3.1-8b-instruct",
+    messages=[{"role": "user", "content": "Describe a futuristic city with advanced technology and green energy solutions."}],
+    temperature=0.2,  # Adjusts creativity
+    max_tokens=100,   # Limits the length of the output
+    top_p=0.7         # Controls diversity through nucleus sampling
+)
+
+# Print the generated response
+print(response.choices[0].message["content"])
+```
+
+This code sends a message to the model and returns an answer based on your input. The `temperature`, `max_tokens`, and `top_p` parameters control the response's creativity, length, and diversity, respectively.
+
+A conversation style may include a default system prompt. You may set this prompt by setting the first message with the role system. For example:
+
+```python
+[
+  {
+  	"role": "system",
+  	"content": "You are Xavier Niel."
+  },
+  {
+  	"role": "user",
+  	"content": "Hello, what is your name?"
+  }
+]
+```
+
+### Model parameters and their effects
+
+The following parameters will influence the output of the model:
+
+- **`messages`**: A list of message objects that represent the conversation history. Each message should have a `role` (e.g., "system", "user", "assistant") and `content`.
+- **`temperature`**: Controls the output's randomness. Lower values (e.g., 0.2) make the output more deterministic, while higher values (e.g., 0.8) make it more creative.
+- **`max_tokens`**: The maximum number of tokens (words or parts of words) in the generated output.
+- **`top_p`**: Recommended for advanced use cases only. You usually only need to use temperature. `top_p` controls the diversity of the output, using nucleus sampling, where the model considers the tokens with top probabilities until the cumulative probability reaches `top_p`.
+- **`stop`**: A string or list of strings where the model will stop generating further tokens. This is useful for controlling the end of the output.
```

Comment:
Pretty advanced too, we may want to add "stop words" in the concepts.

Reply:
Added